mainwindow.cpp

```diff
@@ -42,10 +47,52 @@ void MainWindow::slotPrintMenu()
     while(iterator.hasNext())
     {
         auto item =  iterator.next();
+
+
         item->accept(&visitor);
     }
 }
 
+QJsonArray MainWindow::writeJson(Composite* component)
+{
+    QJsonObject JsonObj;
+    QJsonArray JsonArray;
+    for( int i =0; i < component->subitemsCount(); ++i)
+    {
+        if( component->child(i)->TreeType()=="Menu")
```

Comment:
Maybe I`m not fully understand you task. Is it possible to implement this functionality using  already written visitor pattern. Because checking type "component->child(i)->TreeType()=="Menu"" is not good approach.